src/agent/ICheqd.ts

```diff
@@ -3164,6 +4058,310 @@ export class Cheqd implements IAgentPlugin {
 		});
 	}
 
+	private async BulkUpdateCredentialsWithStatusList(
+		args: ICheqdBulkUpdateCredentialWithStatusListArgs,
+		context: IContext
+	): Promise<BulkBitstringUpdateResult> {
+		// verify credentials, if provided and update options are not
+		if (args?.credentials && !args?.updateOptions) {
+			const verificationResult = await Promise.all(
+				args.credentials.map(async (credential) => {
+					return await context.agent.verifyCredential({
+						...args?.verificationOptions,
+						credential,
+						policies: {
+							credentialStatus: false,
+						},
+					} satisfies IVerifyCredentialArgs);
+				})
+			);
+
+			// early return if verification failed for any credential
+			if (verificationResult.some((result) => !result.verified)) {
+				return {
+					updated: Array(args.credentials.length).fill(false),
+					statusValues: Array(args.credentials.length).fill(BitstringStatusValue.UNKNOWN),
+					statusMessages: Array(args.credentials.length).fill('verification failed'),
+					error: verificationResult.find((result) => !result.verified)!.error || {
+						message: 'verification: could not verify credential',
+					},
+				};
+			}
+		}
+		// validate args - case: credentials
+		if (!args.credentials || !args.credentials.length || args.credentials.length === 0) {
```

Comment:
`args.credentials.length || args.credentials.length === 0` they are redudant can stick to only the latter one here